scripts/fetch-commit-data

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Obtains fix commit statistics for all bug descriptions."""
+import os
+import typing as t
+
+import git
+from robust import BugDescription, ROBUST
+
+
+def print_skip(bug: BugDescription, reason: str) -> None:
+    print(f"Skipping bug: {bug.id} ({reason})")
+
+
+def process_commit(robust: ROBUST,
+                   bug: BugDescription,
+                   dict_commit: t.Dict[str, t.Any]
+                   ) -> None:
+    if 'stats' in dict_commit:
+        raise ValueError("already has 'stats' field")
+
+    try:
+        repo_url = dict_commit['repo']
+        repo_hash = dict_commit['hash']
+    except KeyError as err:
+        raise ValueError(f"commit is mising field: {err}")
+
+    repo = robust.repo(repo_url)
+    try:
+        fix_commit = repo.commit(repo_hash)
+    except (git.exc.BadName, ValueError):
+        raise ValueError(bug, f"bad fix hash: {repo_hash}")
+
+    commit_stats = fix_commit.stats
+    dict_commit['stats'] = {
+        'total': dict(commit_stats.total),
+        'files': {filename: dict(file_stats)
+                  for filename, file_stats in commit_stats.files.items()}
+    }
+    print(dict_commit)
+
+
+def process_bug(robust: ROBUST, bug: BugDescription) -> None:
+    print(f"Processing bug: {bug.id} [{bug.filename}]")
+
+    try:
+        yaml_commits = bug.yaml['fix']['commits']
+    except KeyError as err:
+        print_skip(bug, f"missing field: {err}")
+        return
+
+    if not isinstance(yaml_commits, list):
+        print_skip(bug, "fix.commits is not a list")
+        return
+
+    for yaml_commit in yaml_commits:
+        process_commit(robust, bug, yaml_commit)
+
+    # write to disk
+    bug.save()
+
+
+def main() -> None:
+    dir_robust = os.path.join(os.path.dirname(__file__), '..')
```

Comment:
This script can only be run from the `scripts` dir like this. Is that a limitation we're ok with?

Reply:
The script has to stay in the `scripts` directory, but you should be able to execute it from outside that directory. `os.path.dirname(__file__)` will resolve to the absolute path of the scripts directory, regardless of the current working directory.

Reply:
O right, yes. I need more coffee.